driver.go

```diff
@@ -880,17 +882,42 @@ func (d *Driver) copySSHKeyPair(src string) error {
 	return nil
 }
 
-func (d *Driver) getLocation() (*hcloud.Location, error) {
-	if d.cachedLocation != nil {
-		return d.cachedLocation, nil
+func (d *Driver) getLocations() ([]*hcloud.Location, error) {
+	if len(d.cachedLocations) > 0 {
+		return d.cachedLocations, nil
+	}
+
+	locations := []*hcloud.Location{}
+
+	for _, locationName := range d.Locations {
+		location, _, err := d.getClient().Location.GetByName(context.Background(), locationName)
+
+		if err != nil {
+			return []*hcloud.Location{location}, errors.Wrap(err, "could not get location by name")
```

Comment:
Is there any reason why this returns a non-empty slice with a single (potentially unspecified) element? Right now the call sites will just bail out on error, but returning the last broken element seems odd to me.